### PR TITLE
Major maintenance update 2022-05-30

### PR DESCRIPTION
**Pull request by nuald**
The testing infrastructure has been changed, and it affected the results and the source code:

- changed to use server architecture instead of the desktop one;
- hardened Docker usage to be better suited for the server environment;
- fixed few IPv6 issues as it's a primary protocol in the server environment (however, occasionally IPv4 still used, could be fixed later as it doesn't affect the test results).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -363,77 +366,77 @@ Testing:
...
 Base Docker image: Debian GNU/Linux bookworm/sid
 
 | Language         | Version                         |
 | ---------------- | ------------------------------- |
-| .NET Core        | 6.0.201                         |
-| C#/.NET Core     | 4.1.0-5.22116.13 (dbffaa4a)     |
-| C#/Mono          | 6.12.0.122                      |
+| .NET Core        | 6.0.203                         |
+| C#/.NET Core     | 4.1.0-5.22128.4 (5d10d428)      |
+| C#/Mono          | 6.8.0.105                       |
```
Comment (kostya) [inline README.md:402]: mono version seems downgraded?

File: README.md [hunk 2]
```diff
@@ -223,134 +226,134 @@ have some inaccuracies in number parsing:
...
+|    Nim/gcc (Arraymancer) |    0.066<sub>±0.002</sub> |    5.52<sub>±00.05</sub> + 54.73<sub>±00.11</sub> |      5.46<sub>±00.13</sub> |
+|      C++/clang++ (Eigen) |    0.068<sub>±0.003</sub> |   38.90<sub>±02.59</sub> + 51.51<sub>±02.58</sub> |      5.53<sub>±00.15</sub> |
+|           Python (NumPy) |    0.070<sub>±0.000</sub> |   28.41<sub>±00.12</sub> + 54.96<sub>±00.04</sub> |      6.49<sub>±00.04</sub> |
+|          C++/g++ (Eigen) |    0.070<sub>±0.002</sub> |   29.55<sub>±08.72</sub> + 60.24<sub>±08.76</sub> |      5.13<sub>±00.13</sub> |
+|  Nim/clang (Arraymancer) |    0.072<sub>±0.001</sub> |    6.34<sub>±00.04</sub> + 54.76<sub>±00.08</sub> |      6.07<sub>±00.07</sub> |
+|              Java (ND4J) |    0.081<sub>±0.005</sub> |  104.24<sub>±01.83</sub> + 92.25<sub>±00.16</sub> |      6.35<sub>±00.31</sub> |
+|           Rust (ndarray) |    0.090<sub>±0.000</sub> |    2.63<sub>±00.04</sub> + 68.48<sub>±00.00</sub> |      6.00<sub>±00.03</sub> |
+|       Julia (threads: 4) |    0.191<sub>±0.000</sub> |  235.35<sub>±00.21</sub> + 49.23<sub>±00.90</sub> |     15.54<sub>±00.02</sub> |
```
Comment (kostya) [inline README.md:322]: wonder why julia slow?

File: README.md (hunk 1 above)
Comment (nuald) [reply README.md:402]: Yes, mono custom Debian repository gave me few issues, so I've decided to not use it anymore.

File: README.md (hunk 2 above)
Comment (nuald) [reply README.md:322]: The server has Intel Xeon processor with 4 cores only (https://www.intel.ca/content/www/ca/en/products/sku/212255/intel-xeon-e2324g-processor-8m-cache-3-10-ghz/specifications.html), the previous testing infrastructure used desktop processor with the higher number of cores. OpenBLAS automatically setup the number of threads to use based on the number of CPU cores, so the previous results were using 8 threads, but this one - 4 threads.

**Pull request merged**